### PR TITLE
test: Add more test cases for Base58 parser

### PR DESCRIPTION
**Pull request by ximinez**
## High Level Overview of Change

Adds unit tests related to the AccountID handling updated in 2.2.1 (#5078) 

### Context of Change

2.2.1 was released without all of the relevant unit tests for reasons. This PR finally follows up on that and adds the tests.

(Also fixes an unused variable build warning on Release builds.)

### Type of Change

- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)

### CODE REVIEW

File: include/xrpl/protocol/detail/b58_utils.h [hunk 1]
```diff
@@ -169,8 +169,9 @@ inplace_bigint_div_rem(std::span<uint64_t> numerator, std::uint64_t divisor)
 [[nodiscard]] inline std::array<std::uint8_t, 10>
 b58_10_to_b58_be(std::uint64_t input)
 {
-    constexpr std::uint64_t B_58_10 = 430804206899405824;  // 58^10;
+    static constexpr std::uint64_t B_58_10 = 430804206899405824;  // 58^10;
     assert(input < B_58_10);
+    (void)B_58_10;
```
Comment (mvadari) [inline include/xrpl/protocol/detail/b58_utils.h:174]: What does this line do?

Comment (ximinez) [reply include/xrpl/protocol/detail/b58_utils.h:174]: It fools the compiler into thinking the variable is used, so you don't get unused variable warnings when `assert`s are disabled.
```
In file included from /home/extra/ed/dev/rippled/current/src/ripple/protocol/impl/tokens.cpp:32:
/home/extra/ed/dev/rippled/current/src/ripple/protocol/impl/b58_utils.h: In function ‘std::array<unsigned char, 10> ripple::b58_fast::detail::b58_10_to_b58_be(uint64_t)’:
/home/extra/ed/dev/rippled/current/src/ripple/protocol/impl/b58_utils.h:172:29: warning: unused variable ‘B_58_10’ [-Wunused-variable]
  172 |     constexpr std::uint64_t B_58_10 = 430804206899405824;  // 58^10;
      |                             ^~~~~~~
```
It's just an annoyance, but it was on the branch I had sitting around, so I included it.

**Comment by ximinez** [comment]
The final commit message should not include "NFTs". That function was an unintentional duplicate.

File: src/test/protocol/types_test.cpp [hunk 2]
```diff
@@ -30,6 +30,11 @@ struct types_test : public beast::unit_test::suite
         auto const s = "rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh";
         if (BEAST_EXPECT(parseBase58<AccountID>(s)))
             BEAST_EXPECT(toBase58(*parseBase58<AccountID>(s)) == s);
+        {
```
Comment (vlntb) [inline src/test/protocol/types_test.cpp:33]: [nit] For me the intention is not very clear - is this a deliberate scope introduction or a mistake in using `if` statement. To make it clearer I would mark the borders of the scope of `if` explicitly to make clearly separate for the new scope. 

`
if (BEAST_EXPECT(parseBase58<AccountID>(s))) {
    BEAST_EXPECT(toBase58(*parseBase58<AccountID>(s)) == s);
}

{
    auto const s =
        "âabcd1rNxp4h8apvRis6mJf9Sh8C6iRxfrDWNâabcdAVâ\xc2\x80\xc2\x8f";
    BEAST_EXPECT(!parseBase58<AccountID>(s));
}
`

**Review by vlntb — CHANGES_REQUESTED** [review]
Very minor changes.

File: include/xrpl/protocol/detail/b58_utils.h (hunk 1 above)
Comment (ximinez) [reply include/xrpl/protocol/detail/b58_utils.h:174]: Yeah, I keep forgetting that exists. LOL

File: src/test/protocol/types_test.cpp (hunk 2 above)
Comment (ximinez) [reply src/test/protocol/types_test.cpp:33]: Yeah, now that you mention it, it does look a little off. I changed the test a little bit to make that all clearer, and to remove the need to parse `s` twice.

**Review by vlntb — APPROVED** [review]

**Comment by ximinez** [comment]
Proposed commit message:
```
 test: Add more test cases for Base58 parser (#5174)
```

**Comment by ximinez** [comment]
Hold merging until after 2.3.0 final release.

**Pull request merged**